File: usage/jvmClient/src/main/java/amf/example/creator/RequestBuilder.java
package amf.example.creator;

import amf.model.Request;

/**
 * Builder class to create a request and populate her subclasses with example data.
 */
class RequestBuilder {
    
    static Request albums(){
        final Request request = new Request();
        ParameterHelper.populateIds(request.withQueryParameter("ids"));
        ParameterHelper.populateMaker(request.withQueryParameter("market"));
        
        return request;
    }
    
    static Request album(){
        final Request request = new Request();
        ParameterHelper.populateMaker(request.withQueryParameter("market"));
    
        return request;
    }
    
    
}

File: usage/jsClient/creator/requestBuilder.js
var amf = require('../../../amf-js/target/artifact/amf-module.js');
var ParameterHelper = require('./parameterHelper.js').parameterHelper;

exports.requestBuilder = {

    albums: function () {
        var request = new amf.Request();
        ParameterHelper.populateIds(request.withQueryParameter("ids"))
        ParameterHelper.populateMaker(request.withQueryParameter("market"))

        return request;
    },

    album: function () {
        var request = new amf.Request()
        ParameterHelper.populateMaker(request.withQueryParameter("market"))

        return request;
    }

}